src/components/include/cos_kernel_api.h

```diff
@@ -88,6 +89,8 @@ int cos_sched_rcv(arcvcap_t rcv, thdid_t *thdid, int *blocked, cycles_t *cycles)
 
 int cos_introspect(struct cos_compinfo *ci, capid_t cap, unsigned long op);
 
+int cos_sinv(sinvcap_t sinv);
```

Comment:
perhaps return a `long`, and the 4 arguments can be `unsigned long`.

I choose long because it is identical to the word size in both 32 bit and 64 bit systems.  Thus, it would fully let us use the bits in the arguments to the invocation regardless of platform.